### PR TITLE
Prem AI Templates Llama Index support

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR adds Prem AI prompt template feature support natively in Llama Index.

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py [hunk 1]
```diff
@@ -177,21 +177,57 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
                 all_kwargs.pop(key, None)
         return all_kwargs
 
-    @llm_chat_callback()
-    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
-        all_kwargs = self._get_all_kwargs(**{**self.additional_kwargs, **kwargs})
+    def _prepare_messages_before_chat(
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py:180]: Can you move it to `utils.py` ? You can check [`anthropic`](https://github.com/run-llama/llama_index/blob/00e118c1c1cd5d587c45790b7ed515533279d2b1/llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/utils.py#L54) or any other LLM integration for reference.

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py [hunk 2]
```diff
@@ -177,21 +177,57 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
...
+        all_kwargs = self._get_all_kwargs(**{**self.additional_kwargs, **kwargs})
 
         for message in messages:
             if "system_prompt" in all_kwargs and message.role.value == "system":
                 continue
-
             elif "system_prompt" not in all_kwargs and message.role.value == "system":
                 all_kwargs["system_prompt"] = message.content
-            else:
+            elif message.role.value == "assistant":
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py:191]: You could use `message.role == MessageRole.SYSTEM`

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py [hunk 3]
```diff
@@ -177,21 +177,57 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
...
                 chat_messages.append(
                     {"role": message.role.value, "content": message.content}
                 )
+            elif message.role.value == "user":
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py:195]: You could use `message.role == MessageRole.USER`

**Review by ravi03071991 — CHANGES_REQUESTED** [review]

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py (hunk 2 above)
Comment (Anindyadeep) [reply llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py:191]: Yes, I replaced those with MessageRole

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py (hunk 3 above)
Comment (Anindyadeep) [reply llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py:195]: Done

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py (hunk 1 above)
Comment (Anindyadeep) [reply llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py]: Sure, this makes sense, done

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/utils.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        elif (
+            "system_prompt" not in all_kwargs
+            and message.role.value == MessageRole.SYSTEM
+        ):
+            all_kwargs["system_prompt"] = message.content
+        elif message.role.value == MessageRole.ASSISTANT:
+            chat_messages.append(
+                {"role": message.role.value, "content": message.content}
+            )
+        elif message.role.value == "user":
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/utils.py:20]: `message.role == MessageRole.USER`

**Comment by ravi03071991** [comment]
@Anindyadeep please bump the version in pyproject.toml.

File: llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/utils.py (hunk 4 above)
Comment (Anindyadeep) [reply llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/utils.py]: Done, thanks for pointing out

**Comment by Anindyadeep** [comment]
> @Anindyadeep please bump the version in pyproject.toml.

I added the commits, let me know what else I need to add. Thanks

**Review by ravi03071991 — APPROVED** [review]
LGTM.

**Pull request merged**